build.gradle

```diff
@@ -535,6 +541,24 @@ configure(subprojects - getJsProjects()) { subproject ->
     }
 }
 
+def checkDexer = task('checkDexer', description: 'compiles android class files into dex bytecode', dependsOn: compileKotlin) {
+    getAndroidProjects().each { p->
+        doLast {
+            javaexec {
+                def nameWithoutSuffix = p.name.substring(p.name.indexOf('-') + 1, p.name.lastIndexOf('-'))
+                def jvmProject = prefixedProject(nameWithoutSuffix + '-jvm')
+                def jarToTest = "${jvmProject.projectDir}\\build\\libs\\*.jar"
+                mkdir "d8out/atrium-${nameWithoutSuffix}-${rootProject.version}"
+                classpath = configurations.r8
+                main = "com.android.tools.r8.D8"
+                args = ['--output' , "d8out/atrium-${nameWithoutSuffix}-${rootProject.version}", "${jarToTest}"]
+            }
+        }
+    }
+}
+
+// check.dependsOn(checkDexer)
```

Comment:
don't forget to activate that at the end

Reply:
currently it will fail the build since not all jars dex are successful